coalib/tests/results/SourceAreaTest.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import unittest
+
+sys.path.insert(0, ".")
+from coalib.results.SourceArea import SourcePosition, SourceRange
+
+
+class SourcePositionTest(unittest.TestCase):
+    def test_initialization(self):
+        with self.assertRaises(AssertionError):
+            SourcePosition(None, 0)
+
+        # However this should work:
+        SourcePosition(None, None)
+        SourcePosition("file", None)
+        SourcePosition("file", 4)
+
+    def test_string_conversion(self):
+        uut = SourcePosition("filename", 1)
+        self.assertEqual(str(uut),
+                         "file: 'filename', line: 1, column: None")
+        self.assertRegex(
+            repr(uut),
+            "<SourcePosition object\\(file='filename', line=1, column=None\\) "
+            "at 0x[0-9a-fA-F]+>")
+
+        uut = SourcePosition(None, None)
+        self.assertEqual(str(uut),
+                         "file: None, line: None, column: None")
+        self.assertRegex(
+            repr(uut),
+            "<SourcePosition object\\(file=None, line=None, column=None\\) at "
+            "0x[0-9a-fA-F]+>")
+
+
+class SourceRangeTest(unittest.TestCase):
+    def test_initialization(self):
+        uut = SourceRange(SourcePosition("FileA", "2"))
+        self.assertEqual(uut.end, SourcePosition("FileA", "2"))
+
+    def test_invalid_arguments(self):
+        # arguments must be SourceRanges
+        with self.assertRaises(TypeError):
+            SourceRange(1, SourcePosition("A"))
+
+        with self.assertRaises(TypeError):
+            SourceRange(SourcePosition("A", 2), 1)
+
+    def test_argument_file(self):
+        # both Source_Positions should describe the same file
+        with self.assertRaises(ValueError):
+            SourceRange(SourcePosition("A"), SourcePosition("B"))
+
+    def test_argument_order(self):
+        # end should come after the start
+        with self.assertRaises(ValueError):
+            SourceRange(SourcePosition("A", 2), SourcePosition("A"))
+
+    def test_invalid_comparison(self):
+        with self.assertRaises(TypeError):
+            SourceRange(SourcePosition("B"), SourcePosition("B", 2)) < 1
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expression "SourceRange(SourcePosition('B'), SourcePosition('B', 2)) < 1" is assigned to nothing](https://www.codacy.com/app/lasse/coala/file/1092554918/issues/source?bid=2303100&fileBranchId=2891885#l61)